### PR TITLE
Fix style comment awaiting moderation indentation

### PR DESCRIPTION
**Pull request by cbravobernal**
## What?

Fixed indentation on `Post Comments Block` of the moderation warning message.

## Testing Instructions

1) Create a post. Log out.
2) Add a Comment. Check that the moderation message appears/
3) On trunk, it should have a negative margin left. Screenshot attached.
4) On the branch, it should be aligned with the rest of the comment metadata text. Screenshot attached.

## Screenshots or screencast 

Before:
Screenshot 2022-04-28 at 13 13 35

After:
Screenshot 2022-04-28 at 13 12 21

### CODE REVIEW

**Comment by jasmussen** [comment]
Nice one. I can confirm the issue in trunk:
Screenshot 2022-04-28 at 15 29 08

I wonder if there's a differnt way to solve this, though. Right now the issue appears because there's a 3.25em left padding on the main comment, which is then paired with a negative 3.25em margin later down the stack:
Screenshot 2022-04-28 at 15 29 27

If we remove that initial padding, and the negative margin, the issue appears solved for me simply by virtue of the avatar floating left:
Screenshot 2022-04-28 at 15 30 15

I have a sense that the left padding exists to indent the comment metadata in case it wraps onto multiple lines. In such a case, the subsequent lines would wrap below the avatar, rather than vertically down. However if this is the intent, I don't think we should do it: a hard-coded padding/margin like that won't work if we resize the avatar. Whereas if the avatar is floated and indents naturally, it will work even when resizing. 

What do you think?

One final thought, should we add something like this?

```
.comment-awaiting-moderation {
    margin-top: 1em;
    margin-bottom: 1em;
    display: block;
}
```

That would give some clearance around the message:
Screenshot 2022-04-28 at 15 34 48

**Comment by cbravobernal** [comment]
> What do you think?

I thought about it, but maybe changing that will affect all current Block themes that are using Comments. I guess theme developers handled this issue before, so we may force them to update all themes if we now remove that padding.

Adding the positive margin feels like a patch to be honest, but at least we are not affecting a really common div.

> One final thought, should we add something like this?

That way looks much better.

**Comment by jasmussen** [comment]
> I thought about it, but maybe changing that will affect all current Block themes that are using Comments. I guess theme developers handled this issue before, so we may force them to update all themes if we now remove that padding.

This would only be the case if a block theme used the comments query loop block, and added custom CSS to override parts of this, correct?

If yes, then I wouldn't worry too much about changing that, I think it's important we move towards the best code structure when we can.

Or is this code ported from the classic comments loop?

**Comment by ockham** [comment]
> > I thought about it, but maybe changing that will affect all current Block themes that are using Comments. I guess theme developers handled this issue before, so we may force them to update all themes if we now remove that padding.
> 
> This would only be the case if a block theme used the comments query loop block, and added custom CSS to override parts of this, correct?
> 
> If yes, then I wouldn't worry too much about changing that, I think it's important we move towards the best code structure when we can.
> 
> Or is this code ported from the classic comments loop?

Not sure I'm understanding the question entirely correctly, but this styling is for the ("monolithic") Post Comments block, which is basically a thin wrapper around WP's `comments_template()` function. The block has been around for a while and is used [e.g. in Twenty Twenty-Two](https://github.com/WordPress/wordpress-develop/blob/613cac5a2d809a4078b23945a6c57750bdebd9b0/src/wp-content/themes/twentytwentytwo/templates/single.html#L43).

(It's unrelated to the new "modularized" Comments Query Loop block, although we did find this issue [when working on that block](https://github.com/WordPress/gutenberg/pull/40612#issuecomment-1111884691).)

**Comment by jasmussen** [comment]
Ah, thank you for the clarification. Still, thinking in terms of the styles output by packages/block-library/src/post-comments/style.scss on the frontend, the margin/padding changes could be argued as being in the bugfix category, and would work fine even for TwentyTwentyTwo on updating the plugin or major WordPress version, assuming TT2 or any other theme using it relies on that particular margin and padding for some other custom styles. 

In other words, it seems like the potential breakage of refactoring the code would be minimal, and therefore probably good to fix the issue. Not a strong opinion, but just a general opportunity to improve the code, perhaps!

**Comment by cbravobernal** [comment]
> Ah, thank you for the clarification. Still, thinking in terms of the styles output by packages/block-library/src/post-comments/style.scss on the frontend, the margin/padding changes could be argued as being in the bugfix category, and would work fine even for TwentyTwentyTwo on updating the plugin or major WordPress version, assuming TT2 or any other theme using it relies on that particular margin and padding for some other custom styles.

Then, should we remove the margin and leave everything aligned? That way the theme is responsible for styling the layout.
Screenshot 2022-04-29 at 09 42 52

**Comment by jasmussen** [comment]
Nice! Thank you. I'm unable to test at this point, but the code seems good, and looks like what I tried in the inspector. 👍 👍 on my part

**Review by ockham — APPROVED** [review]
Thanks Carlos and Joen!

**Comment by ockham** [comment]
@adamziel This one would also be great to get into Beta 4, if possible 😊

**Comment by adamziel** [comment]
I cherry picked this change into `wp/6.0` branch: e3ca545674
